Week3/homework/6-step3.js

```diff
@@ -4,3 +4,13 @@ const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
 
 // add your solution here, or add a comment on how you would tackle this problem
+
+const printAllItems = (arr) =>
+    arr.map((item, index) => {
+        return index + ". " + item;
+    }).join('\n');
+
+console.log(printAllItems(arr2d));
+console.log(printAllItems(arr3d));
+
+
```

Comment:
Interesting solution, this approach did not even occur to me.

Does your solution also work with something like this?
```
const arrNd = [[[[1, 2]], [3, 4], [[[5, 6], [7, 8]], [9, 10]], [11, 12]], [[[[[[13, 14]]]]]]];
```
```
> console.log(printAllItems(arrNd))
0. 1,2,3,4,5,6,7,8,9,10,11,12
1. 13,14
```
Yes it does.  Fair enough.  Well done.
I think what was being looked for was creating a recursive function.  Are you able to print the elements out (index is not needed) using recursion?

Reply:
Since in the homework instruction it was not specified that we need to use a recursive function. I tried to come up with something else. :sweat_smile:  Since you and Jim already posted your solutions on Slack, I will just learn from it, because I don't think I can come up with a better solution than yours  at this moment :smile: